framework/src/main/java/org/checkerframework/framework/type/TypeFromMemberVisitor.java

```diff
@@ -30,11 +31,17 @@ public AnnotatedTypeMirror visitVariable(VariableTree node, AnnotatedTypeFactory
         AnnotatedTypeMirror result = TypeFromTree.fromTypeTree(f, node.getType());
 
         // Add primary annotations
-        List<? extends AnnotationTree> annos = node.getModifiers().getAnnotations();
-        if (annos != null && !annos.isEmpty()) {
-            List<AnnotationMirror> ams = TreeUtils.annotationsFromTypeAnnotationTrees(annos);
+        List<? extends AnnotationTree> annoTrees = node.getModifiers().getAnnotations();
+        if (annoTrees != null && !annoTrees.isEmpty()) {
+            List<AnnotationMirror> annos = TreeUtils.annotationsFromTypeAnnotationTrees(annoTrees);
             AnnotatedTypeMirror innerType = AnnotatedTypes.innerMostType(result);
-            innerType.addAnnotations(ams);
+            for (AnnotationMirror anno : annos) {
+                if (AnnotationUtils.isDeclarationAnnotation(anno)) {
```

Comment:
Note how the logic in `addAnnotationsFromElement` is different: it only checks whether the annotation is part of the `org.checkerframework` package, not that it is a declaration annotation.
Shouldn't the logic between these two be the same?

Reply:
No, for elements, the annotation in `@Nullable Object []` is placed in a different location in the element depending on whether it is a type annotation or a declaration annotation. For elements, we only need to move our compatibility annotations to the component type.   

For trees, the annotation in `@Nullable Object []` is the same place in the tree no matter whether the annotation is a declaration annotation or a type annotation.  So we have to move it to the correct place.  

I did forget to treat our compatibility annotations as type annotations, so I've made that change.